VDA-StorageNode-Registry/test/index.test.ts

```diff
@@ -0,0 +1,939 @@
+                ).to.be.revertedWithCustomError(contract, "InvalidRegionCode")
+            })
+
+            it("Failed: Invlaid Latitude",async () => {
+                const invalidLatValues = [-90.05, -180, 91, 500];
+                for (let i = 0; i < invalidLatValues.length; i++) {
+                    const invalidDataCenter = createDatacenterStruct("DataCenter-Test", "US", "North America", invalidLatValues[i], 0 );
+                    await expect(
+                        contract.addDatacenter(invalidDataCenter)
+                    ).to.be.revertedWithCustomError(contract, "InvalidLatitude")
+                }
+            })
+
+            it("Failed: Invalid Longitude",async () => {
+                const invalidLongValues = [-180.1, -270, -400.2523, 181, 360, 500.235];
+                for (let i = 0; i < invalidLongValues.length; i++) {
+                    const invalidDataCenter = createDatacenterStruct("DataCenter-Test", "US", "North America", -90, invalidLongValues[i] );
+                    await expect(
+                        contract.addDatacenter(invalidDataCenter)
+                    ).to.be.revertedWithCustomError(contract, "InvalidLongitude")
+                }
+            })
+
+            it("Success", async () => {
+                for (let i = 0; i < dataCenters.length; i++) {
+                    const tx = await contract.addDatacenter(dataCenters[i])
+
+                    await expect(tx).to.emit(contract, "AddDataCenter");
+
+                    const transactionReceipt = await tx.wait();
+                    const event = transactionReceipt.events?.find(item => {
+                        return item.event === 'AddDataCenter'
+                    })
+                    if (event !== undefined) {
+                        datacenterIds.push(event.args![0])
+                    }
+                }
+            })
+
+            it("Failed : Registered datacenter name", async () => {
+                const invalidDatacenters = [
+                    createDatacenterStruct("Center-1", "US", "North America", -90, -150),
+                    createDatacenterStruct("Center-1", "UK", "Europe", 0, 0),
+                    createDatacenterStruct("Center-2", "AU", "Oceania", -90, -150),
+                    createDatacenterStruct("Center-2", "HK", "Asia", 0, 0),
+                ]
+
+                for (let i = 0; i < invalidDatacenters.length; i++) {
+                    await expect(
+                        contract.addDatacenter(invalidDatacenters[i])
+                    ).to.be.revertedWithCustomError(contract, "InvalidDatacenterName")
+                }
+            })
+
+            it("Can reuse datacenter name once removed", async () => {
+                const currentSnapshot = await takeSnapshot();
+
+                let tx = await contract.removeDatacenter(datacenterIds[0])
+                await expect(tx).to.emit(contract, "RemoveDataCenter").withArgs(datacenterIds[0]);
+
```

Comment:
Add a checkt oconfirm the removed data center can't be fetched with `getDatacenters`